p2p/host/routed/routed.go

```diff
@@ -28,15 +27,12 @@ const AddressTTL = time.Second * 10
 // This allows the Host to find the addresses for peers when
 // it does not have them.
 type RoutedHost struct {
-	host  host.Host // embedded other host.
-	route Routing
+	host.Host
+	route routing.PeerRouting
 }
```

Comment:
Do we need to change route from Routing to routing.PeerRouting? 

The rest of the change looks good to me.

Reply:
SGTM.